src/lib/OS/PageTable.cc

```diff
@@ -0,0 +1,147 @@
+#include "simeng/OS/PageTable.hh"
+
+#include <cmath>
+#include <iostream>
+#include <vector>
+
+#include "simeng/util/Math.hh"
+
+namespace simeng {
+namespace OS {
+
+bool PageTable::isMapped(uint64_t vaddr) { return find(vaddr) != table_.end(); }
+
+uint64_t PageTable::generateOffsetMask(uint64_t pageSize) {
+  if (!isPow2(PAGE_SIZE)) {
```

Comment:
You're passing pageSize into the function, yet checking that the constExpr is a power of 2. Please make this consistent to the value you are using

Reply:
done